File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/trees.java
/**
 * Java Implementation: Trees (Binary Tree & Traversals)
 * Invariant: Traversals O(N), Depth O(log N) to O(N)
 */
package seed.dsa.core;

import java.util.*;

public class TreesBinaryTreeTraversals<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/csharp/trees.cs
/**
 * C# Implementation: Trees (Binary Tree & Traversals)
 * Invariant: Traversals O(N), Depth O(log N) to O(N)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class TreesBinaryTreeTraversals<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}
